src/opnsense/service/templates/OPNsense/Dnsmasq/dnsmasq.conf

```diff
@@ -29,9 +29,20 @@ no-dhcp-interface={{helpers.physical_interfaces(dnsmasq.dhcp.no_interface.split(
 dhcp-lease-max={{dnsmasq.dhcp.lease_max}}
 {% endif %}
 
-{% if dnsmasq.dhcp.fqdn == '1' %}
+{% if helpers.exists('dnsmasq.dhcp_ranges') and dnsmasq.dhcp.fqdn == '1' %}
```

Comment:
well this looks a bit odd now. you always want to enable it, move the first conditional into the loop below...

Reply:
(the idea mentioned played on the fact that maybe you don't want to offer dhcp.fqdn setting at all)